sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_deserialize.py

```diff
@@ -139,7 +150,7 @@ def normalize_headers(headers):
     return normalized
 
 
-def process_storage_error(storage_error) -> NoReturn:  # pylint:disable=too-many-statements
+def process_storage_error(storage_error) -> NoReturn:  # type: ignore [misc] # pylint:disable=too-many-statements
```

Comment:
There's already a `process_storage_error` in `_shared`'s `response_handlers.py`. The other packages use that instead of directly having a `process_storage_error` in `_deserialize.py`. Worth moving all `process_storage_error` from `_deserialize.py` in datalake to shared folder's `process_storage_error`?